stakingModel/test/GoodFundManager.e2e.test.js

```diff
@@ -0,0 +1,144 @@
+    console.log(deploy_settings);
+    // schemes
+    setReserve = await FundManagerSetReserve.new(avatarAddress, goodFundManager.address, goodReserve.address);
+  });
+
+  it("should mints cdai and updates exchange rate", async () => {
+    await dai.mint(staker, web3.utils.toWei("100", "ether"));
+    await dai.approve(simpleStaking.address, web3.utils.toWei("100", "ether"), {
+      from: staker
+    });
+    await simpleStaking
+    .stakeDAI(web3.utils.toWei("100", "ether"), {
+      from: staker
+    })
+    .catch(console.log);
+    await cDAI.exchangeRateCurrent();
+    let rate = await cDAI.exchangeRateStored();
+    expect(rate.toString()).to.be.equal("10201010101010101010101010101");
+  });
+
+  it("should be able to set the reserve", async () => {
+    const executeProposalEventExists = await proposeAndRegister(setReserve.address,
+      registrar, proposalId, absoluteVote, avatarAddress, founder);
+    // Verifies that the ExecuteProposal event has been emitted
+    expect(executeProposalEventExists).to.be.true;
+    await setReserve.setReserve();
+    const reserve1 = await goodFundManager.reserve();
+    expect(reserve1).to.be.equal(goodReserve.address)
+  });
+
+  it("should not mint UBI if not in the interval", async () => {
+    const error = await goodFundManager
+      .transferInterest(simpleStaking.address)
+      .catch(e => e);
+    expect(error.message).to.have.string("wait for the next interval");
+  });
+
+  it("should collect the interest and transfer it to the reserve and recieves minted gd back to the staking contract", async () => {
+    await next_interval();
+    await cDAI.exchangeRateCurrent();
+    let recipientBefore = await goodDollar.balanceOf(ubiBridgeRecipient);
+    await goodFundManager.transferInterest(simpleStaking.address);
+    let recipientAfter = await goodDollar.balanceOf(ubiBridgeRecipient);
+    let stakingGDBalance = await goodDollar.balanceOf(simpleStaking.address);
+    expect(stakingGDBalance.toString()).to.be.equal("0"); //100% of interest is donated, so nothing is returned to staking
+    expect(recipientAfter.sub(recipientBefore).toString()).to.be.equal("1904085"); //970492 interest + 594 minted from expansion
+  });
+
+  it("should keep the same price while collects interest and recieves minted gd back to the staking contract", async () => {
+    await next_interval();
+    await cDAI.exchangeRateCurrent();
+    const gdPriceBefore = await marketMaker.currentPrice(cDAI.address);
+    await goodFundManager.transferInterest(simpleStaking.address);
+    const gdPriceAfter = await marketMaker.currentPrice(cDAI.address);
+    expect(
+      Math.floor(gdPriceAfter.toNumber() / 100).toString()
+    ).to.be.equal(Math.floor(gdPriceBefore.toNumber() / 100).toString());
+  });
+
+  it("should create be able to stake dai", async () => {
+    await next_interval();
+    const stakingMock = await SimpleDAIStakingMock.new(
```

Comment:
you are not supposed to deploy ANY contracts, this is e2e tests, you test the contracts deployed in the migration scripts

Reply:
the test moved to GoodFundManager.test.js
deadb071dc8af2d0955c9eafcdb8c31b3fd7556e